client/src/consumer.rs

```diff
@@ -366,16 +372,27 @@ impl<'a> ConsumerBuilder<'a, Set<Regex>, Set<String>, Set<SubType>> {
     }
 }
 
+/// Details about the current state of the Consumer
+#[derive(Debug, Clone)]
+pub struct ConsumerState {
+    pub connected_topics: Vec<String>,
```

Comment:
Since this struct is cloned for every message received, I'm wondering if it might make sense to use something like an `Arc<[String]>` to avoid unnecessary allocation when creating one.

For a small number of topics, you probably won't notice, but if there were like 100+ topics on the consumer that would probably start to be noticeable.